agent/translate_addr.go

```diff
@@ -11,7 +11,7 @@ import (
 // parameter is the dc the datacenter this node is from.
 func (a *Agent) TranslateServicePort(dc string, port int, taggedAddresses map[string]structs.ServiceAddress) int {
 	if a.config.TranslateWANAddrs && (a.config.Datacenter != dc) {
-		if wanAddr, ok := taggedAddresses["wan"]; ok && wanAddr.Port != 0 {
+		if wanAddr, ok := taggedAddresses[structs.TaggedAddressWAN]; ok && wanAddr.Port != 0 {
```

Comment:
Same question here? Should we sometimes use the `TaggedAddressWANIPv6` or `TaggedAddressWANIPv4` entries in the map?

I am thinking we need a way similar to the `BestAddress` method on the `CheckServiceNode` type that will select an address given parameters like:

1. Do we need a WAN address?
2. Do we support IPv4 IPv6 or both?

Reply:
Im not sure about this, there is currently no way to specify if v6 is preferred over v4 using the HTTP api. I implemented this logic in the DNS part because A is different from AAAA. But here what should we do if the service both has a WanIPv4 and a WanIPv6 set ?

Reply:
This is probably fine for now. We could in the future figure out if the HTTP request is being made over IPv6 and default to that (or provide a query param or header to allow indicating the preferred IP protocol)

That is really outside the scope of this PR though.